pwnlib/shellcraft/templates/amd64/darwin/cat2.asm

```diff
@@ -0,0 +1,15 @@
+<%
+  from pwnlib.shellcraft.amd64 import syscall, pushstr
+  from pwnlib.shellcraft import common
+%>
+<%page args="filename, fd=1, length=0x4000"/>
+<%docstring>
+Opens a file and writes its contents to the specified file descriptor.
+Uses an extra stack buffer and must know the length.
+</%docstring>
+
+    ${pushstr(filename)}
+    ${syscall('SYS_open', 'rsp', 'O_RDONLY', length)}
+    sub rsp, rdx
+    ${syscall('SYS_read', 'rax', 'rsp', 'rdx')}
+    ${syscall('SYS_write', fd, 'rsp', 'rax')}
```

Comment:
Now I think we can shave off a few extra bytes:
```suggestion
    ${syscall('SYS_write', fd, 'rsi', 'rax')}
```

Reply:
this was broken anyway, fixed now. Please check